crates/flycheck/src/lib.rs

```diff
@@ -153,14 +157,24 @@ impl FlycheckActor {
         while let Some(event) = self.next_event(&inbox) {
             match event {
                 Event::Restart(Restart) => {
+                    // Drop and cancel the previously spawned process
+                    self.cargo_handle.take();
```

Comment:
```suggestion
                    
                    if let Some(handle) = self.cargo_handle.take() {
                        // Drop and cancel the previously spawned process
                        handle.kill();
                        let _ = handle.wait();
                    }
```

Comment's great, but I bet explicit code is event better!